src/client/ua_client.c

```diff
@@ -291,17 +295,26 @@ static UA_StatusCode SecureChannelHandshake(UA_Client *client, UA_Boolean renew)
 
     retval = response.responseHeader.serviceResult;
     if(retval == UA_STATUSCODE_GOOD) {
+        /* Does the sequence number match? */
+        retval = UA_SecureChannel_checkSequenceNumber(seqHeader.sequenceNumber, client->channel);
+        if (retval != UA_STATUSCODE_GOOD){
+            UA_LOG_INFO_CHANNEL(client->config.logger, client->channel,
+                                "The sequence number was not increased by one. Got %i, expected %i",
+                                seqHeader.sequenceNumber, client->channel->receiveSequenceNumber + 1);
+            return UA_STATUSCODE_BADINTERNALERROR;
```

Comment:
Maybe this can be folded into one clean-up.
Just add
```c
retval |= UA_SecureChannel_checkSequenceNumber(seqHeader.sequenceNumber, client->channel);
```
or similar before line 286.